api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy

```diff
@@ -88,9 +110,68 @@ class IDXAuthenticationWrapperTest {
+        assertThat(newUserRegistrationResponse.getErrors(), empty())
+        assertThat(newUserRegistrationResponse.getFormValues(), notNullValue())
+
+        setMockResponse(requestExecutor, "interact", "interact-response", 200, MediaType.APPLICATION_JSON)
+        setMockResponse(requestExecutor, "introspect", "enroll-user-response", 200, MediaType.APPLICATION_JSON)
+        setMockResponse(requestExecutor, "enroll/new", "enroll-profile-error-response", 400, mediaTypeAppIonJson)
+
+        AuthenticationResponse authenticationResponse =
+                idxAuthenticationWrapper.register(newUserRegistrationResponse.getProceedContext(), getUserProfile())
+        assertThat(authenticationResponse.getAuthenticators(), nullValue())
+        assertThat(authenticationResponse.getAuthenticationStatus(), nullValue())
+        assertThat(authenticationResponse.getErrors(), notNullValue())
+        assertThat(authenticationResponse.getErrors(), hasItems(
+                "Provided value for property 'Email' does not match required pattern",
+                "'Email' must be in the form of an email address"
+        ))
+    }
+
+    @Test
+    void verifyEmailSuccessResponseTest() {
+
+        def requestExecutor = mock(RequestExecutor)
+        def idxClient = new BaseIDXClient(getClientConfiguration(), requestExecutor)
+        def idxAuthenticationWrapper = new IDXAuthenticationWrapper()
+        //replace idxClient with mock idxClient
+        setInternalState(idxAuthenticationWrapper, "client", idxClient)
+
+        setMockResponse(requestExecutor, "interact", "interact-response", 200, MediaType.APPLICATION_JSON)
+        setMockResponse(requestExecutor, "introspect", "introspect-response", 200, mediaTypeAppIonJson)
+
+        AuthenticationTransaction introspectTransaction = AuthenticationTransaction.create(idxClient)
+        VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions("wrong_code")
+
+        setMockResponse(requestExecutor, "introspect", "challenge-response", 200, mediaTypeAppIonJson)
+        setMockResponse(requestExecutor, "challenge/answer", "challenge-response", 200, mediaTypeAppIonJson)
+
+        AuthenticationResponse authenticationResponse = idxAuthenticationWrapper.verifyAuthenticator(
+                new ProceedContext(introspectTransaction.clientContext,
+                        introspectTransaction.getStateHandle(), "/challenge/answer", null),
+                verifyAuthenticatorOptions
+        )
 
-        setStubbedInteractResponse(requestExecutor)
-        setStubbedEnrollAuthenticatorResponse(requestExecutor)
+        assertThat(authenticationResponse, notNullValue())
+        assertThat(authenticationResponse.getErrors(), empty())
+        assertThat(authenticationResponse.getAuthenticationStatus(),
+                is(AuthenticationStatus.AWAITING_AUTHENTICATOR_VERIFICATION))
```

Comment:
Why is it `AuthenticationStatus.AWAITING_AUTHENTICATOR_VERIFICATION`, I'd think we would have moved on after verifying.

Reply:
yes, you are right! should be AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION